selfdrive/hardware/tici/agnos.py

```diff
@@ -6,12 +6,13 @@
 import struct
 import subprocess
 import os
+from typing import Generator
 
 from common.spinner import Spinner
 
 
 class StreamingDecompressor:
-  def __init__(self, url):
+  def __init__(self, url: str) -> None:
```

Comment:
do we really want to annotate __init__?

Reply:
I'm not sure how standard the return type is (I've seen it quite a bit), but yea I would definitely type hint the params, helps like any other function (this example seems pretty trivial since it's just a url but it'd be more important in other situations with many possibly confusing params).
![image](https://user-images.githubusercontent.com/17343631/120121483-bc53b600-c171-11eb-81f2-df41e9679f8b.png)

Reply:
Turns out this return value type hinting is actually a bit of a rabbit hole with [this](https://github.com/python/mypy/issues/604#issuecomment-348525995) being the implemented conclusion.

Reply:
So if I read that issue correctly, we don't need to annotate the return type since we annotate url and the `-> None` is implied?

Reply:
Yeah, correct. Personally I find this pretty unobvious and being that the second line of the [Zen of Python](https://www.python.org/dev/peps/pep-0020/) reads "Explicit is better than implicit.", this seems pretty unpythonic lol.

I type hint everything in my codebase personally (so I would keep the ` -> None` in this case), but please tell me which you prefer and I'll keep that standard while type hinting openpilot :) cheers

Reply:
Leaving `None` is fine with me.